Assignment 3

Shiraz-Latif opened the pull request:
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Submission of Assignment-3

## What did you learn from the changes you have made?
Model validation 
hyperparameters tuning 
evaluation of model 
model selection 

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
use of functions to avoid repeated tasks 

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
pipeline built was fine but fitting was generated alot of errors, ultimately was able to figure out the solution. 
Still issue with the model assessment, generating nan in the score. 

## How were these changes tested?
Testing was performed by fitting individual transformation or pipeline 

## A reference to a related issue in your repository (if applicable)

## Checklist
- [x] I can confirm that my changes are working as intended

Shiraz-Latif committed:
Assignment-1

Shiraz-Latif committed:
Assignment-2 submission

Shiraz-Latif committed:
Assignment-3-a

ananyajha2000 reviewed (commented):
Marked as incomplete
There are a lot of errors. Please redo. Some of the issues:
1) X_train, X_test etc should be defined before Preprocessing
2) preproc1 = ColumnTransformer(
    transformers=[
        ('num_trans', StandardScaler(), num_feat), 
        ('cat_trans', OneHotEncoder(handle_unknown='infrequent_if_exist'), #unknown categories should be explicitly handled
                                         cat_feat)])
3) Code is commented out when it is required 
4) Explain not answered

Shiraz-Latif committed:
Assignment-3 Resubmission

Shiraz-Latif commented:
@ananyajha2000  Thanks for reviewing the work. I reworked on the identified issues and fixed all. 

Please have a look and let me know if there is anything that need further attention. 

Thanks

ananyajha2000 commented:
A3 marked complete

Shiraz-Latif commented:
Thanks @ananyajha2000 for marking the assignment. 
Can you please also mark my Assignment-1. I can see it is still not marked. 
Thanks 

https://github.com/Shiraz-Latif/scaling_to_production/pull/1
